daemon/cmd/daemon_main.go

```diff
@@ -1534,6 +1537,11 @@ func initKubeProxyReplacementOptions() {
 		return
 	}
 
+	// Disable the HealthCheckNodePort in partial mode when kube-proxy is running
+	if option.Config.KubeProxyReplacement == option.KubeProxyReplacementPartial {
+		option.Config.EnableHealthCheckNodePort = false
```

Comment:
I think in the partial mode you want to use user's provided value instead of hardcoding it.

Reply:
Ok, will address it.

Reply:
In that case, I don't think we need this check at all.

Reply:
Then we should auto-enable it in the `strict` and `probe` modes.

Reply:
> Then we should auto-enable it in the `strict` and `probe` modes.

In strict and probe modes, the default value is 'true' and that would be sufficient. What do you think? Will that not kick in.

Reply:
In strict mode we forcefully enable all the kube-proxy replacement features (incl. the healthcheck nodeport), and in probe mode we probe for what can be enabled. In both cases, the flags for individual features are ignored. So, we need to auto-enable for both.

Reply:
Done.